ngraph/core/src/op/non_max_suppression.cpp

```diff
@@ -856,3 +1048,64 @@ namespace ngraph
         return s << as_string(type);
     }
 } // namespace ngraph
+
+bool op::v5::NonMaxSuppression::evaluate(const HostTensorVector& outputs,
+                                         const HostTensorVector& inputs) const
+{
+    int64_t max_output_boxes_per_class = max_boxes_output_from_input();
+    float iou_threshold = iou_threshold_from_input();
+    float score_threshold = score_threshold_from_input();
+    float soft_nms_sigma = soft_nms_sigma_from_input();
+
+    auto selected_indices_shape = infer_selected_indices_shape(inputs, max_output_boxes_per_class);
+    Shape out_shape = selected_indices_shape.to_shape();
+
+    Shape boxes_shape = inputs[boxes_port]->get_shape();
+    Shape scores_shape = inputs[scores_port]->get_shape();
+
+    auto boxes_data = prepare_boxes_data(inputs[boxes_port], boxes_shape, m_box_encoding);
+    auto scores_data = prepare_scores_data(inputs[scores_port], scores_shape);
+
+    size_t out_shape_size = shape_size(out_shape);
+
+    std::vector<int64_t> selected_indices(out_shape_size);
+    std::vector<float> selected_scores(out_shape_size);
+    int64_t valid_outputs = 0;
+
+    runtime::reference::non_max_suppression(boxes_data.data(),
+                                            boxes_shape,
+                                            scores_data.data(),
+                                            scores_shape,
+                                            max_output_boxes_per_class,
+                                            iou_threshold,
+                                            score_threshold,
+                                            soft_nms_sigma,
+                                            selected_indices.data(),
+                                            out_shape,
+                                            selected_scores.data(),
+                                            out_shape,
+                                            &valid_outputs,
+                                            m_sort_result_descending);
+
+    outputs[0]->set_element_type(m_output_type);
+    outputs[0]->set_shape(Shape{static_cast<size_t>(valid_outputs), 3});
+
+    size_t num_of_outputs = outputs.size();
+
+    if (num_of_outputs >= 2)
+    {
+        outputs[1]->set_element_type(element::f32);
```

Comment:
Why do you hardcode element type to f32? As I know HostTensor always has some element type.

Reply:
Fixed.